File: 🔴 Hard/995 - Minimum Number of K Consecutive Bit Flips/_995MinNoOfKConsBitFlips.java
class Solution {
    public int minKBitFlips(int[] nums, int k) {
        int n = nums.length;
        int w = 0;
        int count = 0;
        boolean[] flipped = new boolean[n];

        for (int i = 0; i < n; i++) {
            if (i >= k) {
                if (flipped[i - k]) {
                    w--;
                }
            }
            if (w % 2 == nums[i]) {
                if (i + k > n) {
                    return -1;
                }

                w++;
                flipped[i] = true;
                count++;
            }
        }

        return count;
    }
}

File: 🔴 Hard/995 - Minimum Number of K Consecutive Bit Flips/_995MinNoOfKConsBitFlips.py
class Solution:
    def minKBitFlips(self, nums: List[int], k: int) -> int:
        n = len(nums)
        w = 0
        count = 0
        flipped = [False] * n

        for i in range(n):
            if i >= k:
                if flipped[i - k]:
                    w -= 1

            if w % 2 == nums[i]:
                if i + k > n:
                    return -1

                w += 1
                flipped[i] = True
                count += 1
        return count
